explorer/ast/static_scope.cpp

```diff
@@ -67,6 +79,14 @@ auto StaticScope::ResolveHere(std::optional<ValueNodeView> this_scope,
       return ProgramError(source_loc)
              << "name '" << name << "' has not been declared in this scope";
     }
+  } else {
+    if (this_scope) {
+      if (trace_stream_->is_enabled()) {
+        *trace_stream_ << "--- declared `" << name << "` in "
+                       << PrintAsID(this_scope->base()) << "` ("
+                       << this_scope->base().source_loc() << ")\n";
```

Comment:
This function is used when looking / resolving up an already-declared name, not when declaring a new name. So I think we should produce a message that says something more like this:

```
--- resolved `foo` at myfile.carbon:5 in `fn F` as `var Foo: i32`
```

Reply:
Okay, thanks for letting me know about that. Can we put the source location at end of line like this?
```
--- resolved `foo` in `fn F` as `var Foo: i32` at (.../myfile.carbon:5)
```
I was thinking of keeping source location at the end of line for consistency.